### PR TITLE
Feat/agent discovery and mcp consent

### PR DESCRIPTION
**Pull request by mitcheman**
This pull request updates and expands the test suites for agent validation and file transcription endpoints, focusing on supporting new agent ID/name formats, improving test clarity, and cleaning up outdated or redundant test cases. The most significant changes are grouped below.

**Agent Validation Endpoint Test Improvements**
- Updated mocks and tests to use `AIProjectClient` from `@azure/ai-projects` instead of `AgentsClient` from `@azure/ai-agents`, reflecting backend changes. [[1]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L16-R16) [[2]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L31-R32) [[3]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L58-R69) [[4]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L254-R308) F59e11e7L567R567, [[5]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L589-R674)
- Expanded validation tests to cover both legacy agent IDs (e.g., `asst_...`) and new agent name formats (e.g., `my-agent`, `gpt-41`), including edge cases like hyphens, underscores, and mixed case. [[1]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L45-R46) [[2]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L154-R160) [[3]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L166-R174) [[4]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0R183-R201) [[5]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L209-R213) [[6]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L221-R267) [[7]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0R320-R339) F59e11e7L346R346, F59e11e7L374R374, F59e11e7L391R391, F59e11e7L543R543, [[8]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L559-R645)
- Improved error handling and messaging in tests for invalid agent IDs and missing agents. [[1]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L154-R160) [[2]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L166-R174) [[3]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0R183-R201) F59e11e7L374R374, F59e11e7L391R391)
- Added integration-style tests for full validation flows with both legacy and new agent IDs. [[1]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L489-R546) [[2]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L513-R594) [[3]](diffhunk://#diff-1f61e74d08a4e9b311e2a6ce760bbe178bbc31331370e23ecb481ec8669e10e0L559-R645)

**File Transcription Endpoint Test Updates**
- Changed temp file naming test to expect a timestamp (from UUID), aligning with implementation changes.
- Updated and simplified file system mocks for clarity and reliability.
- Removed redundant or outdated edge case tests, such as those for file size limits and path separator rejection, to streamline the suite. [[1]](diffhunk://#diff-15e8926695443dd779c1a0996b47da8061cb4b5567c11a08403c197861da1059L394-L404) [[2]](diffhunk://#diff-15e8926695443dd779c1a0996b47da8061cb4b5567c11a08403c197861da1059L485-L516)
- Clarified and renamed tests for handling file IDs with special characters.

**Other Test Suite Refinements**
- Updated file ID validation tests to clarify accepted file extension lengths.

These changes improve test coverage for new agent ID formats, ensure alignment with backend updates, and clean up unnecessary or outdated test cases.

### CODE REVIEW

File: client/handlers/chatInput/image-upload.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    // Update status to completed
+    setFilePreviews((prevFilePreviews) =>
+      prevFilePreviews.map((preview) =>
+        preview.name === file.name
+          ? { ...preview, status: 'completed' }
+          : preview,
+      ),
+    );
+  } catch (error) {
+    console.error('Image upload failed:', error);
```
[the remark this answers is not in the document]
Comment (blazeiburgess) [reply client/handlers/chatInput/image-upload.ts]: Client-side `console.error` (browser devtools only) — no server log sink to forge, so not exploitable. `file.name` is already sanitized into `safeName` for the user-facing toast.
